src/main/java/io/lakefs/routerfs/RouterFileSystem.java

```diff
@@ -135,7 +148,8 @@ public boolean rename(Path src, Path dst) throws IOException {
      */
     @Override
     public boolean delete(Path f, boolean recursive) throws IOException {
-        return false;
+        FileSystemPathTuple tuple = generateFSPathTuple(f);
+        return tuple.getFileSystem().delete(tuple.getPath(), recursive);
```

Comment:
Let's mark with a bug that this fails when recursively deleting across a filesystem.  It probably does not matter, but let's document it.